libs/zip/extract.go

```diff
@@ -0,0 +1,43 @@
+package zip
+
+import (
+	"archive/zip"
+	"io"
+	"io/fs"
+	"os"
+	"path/filepath"
+)
+
+func Extract(src string, dst string) error {
+	zipReader, err := zip.OpenReader(src)
+	if err != nil {
+		return err
+	}
+	defer zipReader.Close()
+
+	return fs.WalkDir(zipReader, ".", func(path string, d fs.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+
+		targetPath := filepath.Join(dst, path)
+		if d.IsDir() {
+			return os.MkdirAll(targetPath, os.ModePerm)
```

Comment:
This is 0o777 which is too permissive. Please use 0o755 so that the writable bit is only set for owners.

Reply:
done